package/MDAnalysis/coordinates/TRZ.py

```diff
@@ -221,6 +226,14 @@ def __init__(self, trzfilename, n_atoms=None, **kwargs):
 
         self._read_next_timestep()
 
+        try:
+            self._get_dt()
+        except OSError:
+            raise ValueError("Supplied n_atoms {} is incompatible "
+                             "with provided trajectory file. "
+                             "Maybe `topology` is wrong?".format(self.n_atoms))
+
+
```

Comment:
It is not clear why `_get_dt()` will fail with the wrong `n_atom`. What's the stack trace and where does the ValueError come from? (I assume this relies on being able to do `self.next()`, which actually fills the `ts`.... but then why does `self._read_next_timestep()` on the preceding line work? 🤷 )

This might work and fix an error but I think we should rather check sooner and in a more transparent manner if at all possible.

Reply:
`_read_next_timestep` (used by `__next__`) only fills the `ts` by reading a chunk from the TRZ file using `np.fromfile` with the provided `frame_contents` (defined by `n_atoms`) no matter it is right or wrong.

`read_frame`, and other related functions (e.g. `_get_dt` will go next and read frame back), on the other hand, will use `self._seek`. With an abnormal `trajectory.next().frame`, This will always reach an error.